src/mjlab/entity/entity.py

```diff
@@ -125,7 +125,8 @@ def __init__(self, cfg: EntityCfg) -> None:
       self._non_free_joints = tuple(all_joints[1:])
 
     self._apply_spec_editors()
-    self._add_initial_state_keyframe()
+    if not self._spec.key("init_state"):
+        self._add_initial_state_keyframe()
```

Comment:
Note that this is a workaround and should be discussed, I might have misunderstood something
This is here because `key.ctrl = joint_pos` below seems to assume `ctrl` and `joint_pos` have the same size